### PR TITLE
PowerVM documentation update

### PR DESCRIPTION
**Pull request by abalib**

### CODE REVIEW

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+-------------------------------------
+
+- Each VAS window will be configured with a number of credits limiting
+  the number of nx-gzip requests that can be simultanenously sent
+  through window, administrator configurable. On PowerNV, 1024 credits
+  are configured per window.  Whereas on PowerVM, by default 1 credit
+  per window are allowed
+
+- PowerVM introduces 2 different types of credits called the Default
+  and QoS credits Default uses the normal priority nx-gzip queue. QoS
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:132]: Missing a period (or colon) after QoS credits.

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  specific VAS instance.  Instead, the PowerVM hypervisor manages
+  allocations of VAS instances. The H_ALLOCATE_VAS_WINDOW hcall allows
+  selection by domain identifiers (H_HOME_NODE_ASSOCIATIVITY values by
+  cpu). By default, the hypervisor selects VAS instance closer to CPU
+  resources that the partition uses. Therefore, the vas_id field in
+  the ioctl interface is ignored on PowerVM except for vas_id=-1 which
+  is used to allocate a VAS window based on CPU that the process is
+  executing. This option is needed for process affinity to NUMA node.
+
+  PowerVM also enforces hard limits on nx-gzip data lengths, 1MB by
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:158]: I think it should start with a `-` as the other items.

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  support multiple outstanding requests in any case. For single
+  threaded applications practically the performance will be same.  For
+  multiple threads sharing a window, the requests will not be queued
+  in the nx-gzip queue but the application, e.g., libnxz, must retry
+  rejected requests.
+
+- Two different types of credit:
+
+  Quality of service (QoS) and Default QoS credits are configured in
+  HMC. QoS credits quarantee that these credits are always available
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:193]: s/quarantee/guarantee/

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+/sys/kernel/vas/VasCaps/VQosGzip    (Called QoS GZIP capabilities)
+	avail_lpar_creds   target_lpar_creds  used_lpar_creds
+ 
+
+  VAS instances: There is no concept of VAS ID on PowerVM. Instead the
+  PowerVM must receive a domain ID (based on the CPU ID) in the
+  H_ALLOCATE_VAS_WINDOW hcall to allocate a VAS window on the specific
+  chip and node.  Domain ID to VAS ID mapping cannot be easily
+  determined by the user-space. Therefore, in the Linux driver
+  implementation When user-space sets vas_id= -1, the domain ID is set
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:238]: s/When/when/

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+1048576
+
+Max request buffer length is enforced. Minimums are not enforced.
+
+
+Page Faults
+-----------
+
+  Handling of nx-gzip caused page faults is same between PowerVM and
+  PowerN. Refer to the ERR_NX_AT_FAULT condition code in the libnxz
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:271]: s/PowerN/PowerNV/

File: doc/Addendum-NX-GZIP-for-PowerVM.txt [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+  capabilities must be updated.
+
+  Debugfs: provide window status and information through debugfs
+
+
+References
+----------
+
+[P1]
+https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=238878
```
Comment (rzinsly) [inline doc/Addendum-NX-GZIP-for-PowerVM.txt:295]: The link is empty, it should be https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=238878&state=*

**Review by rzinsly — CHANGES_REQUESTED** [review]
The content looks good, there is some typos and punctuation missing. Some paragraphs end with a period and others not, it's better to end all the same way.

File: doc/Addendum-NX-GZIP-for-PowerVM.txt (hunk 1 above)
Comment (abalib) [reply doc/Addendum-NX-GZIP-for-PowerVM.txt]: Thank you

**Review by rzinsly — APPROVED** [review]
LGTM but I would prefer both commits squashed.

**Comment by abalib** [comment]
> LGTM but I would prefer both commits squashed.

OK I did something to squash but I still see two commits.  What goes?

**Pull request closed without merging**